lib/dictionary.test.js

```diff
@@ -78,21 +77,69 @@ describe("when I'm using a dictionary", () => {
 
     describe("and I add a dictionary with names and weights", () => {
         beforeEach(() =>{
-            dictionary.addEntry({testNames: ["test", new WeightedEntry("trial", 3), "exam"]});
+            dictionary.addEntry({testNames: ["test", {value: "trial", weight: 3}, "exam"]});
         });
 
-        it("should insert the entries by their weights", () => {
+        it("should get the entries by their weights", () => {
             let entry = dictionary.getEntry("testNames");
-            let processWeights = dictionary.processWeights(entry);
-            expect(processWeights.length).toBe(5);
-            expect(processWeights[0]).toBe("test");
-            expect(processWeights[1]).toBe("trial");
-            expect(processWeights[2]).toBe("trial");
-            expect(processWeights[3]).toBe("trial");
-            expect(processWeights[4]).toBe("exam");
+            expect(entry.length).toBe(5);
+            expect(entry[0]).toBe("test");
+            expect(entry[1]).toBe("trial");
+            expect(entry[2]).toBe("trial");
+            expect(entry[3]).toBe("trial");
+            expect(entry[4]).toBe("exam");
         });
     });
 
+    describe("and I add a dictionary with names and weights nested object", () => {
+        beforeEach(() =>{
+            dictionary.addEntry({world:{continents:{europe:{countries:["England",{value: "America", weight: 3},"Russia"], currencies:["$"]}}}});
+        });
+
+        it("should get the entries by their weights", () => {
+            let entry = dictionary.getEntry("world.continents.europe.countries");
+            expect(entry.length).toBe(5);
+            expect(entry[0]).toBe("England");
+            expect(entry[1]).toBe("America");
+            expect(entry[2]).toBe("America");
+            expect(entry[3]).toBe("America");
+            expect(entry[4]).toBe("Russia");
+        });
+    });
+
+    describe("and I add contextual entries to the dictionary that dont match the current contexts", () => {
```

Comment:
shouldn't there be a test here to check it selects if there is a match?

Reply:
seeing the problem with testing this. Although it does make me wonder if contexts should be limited to the entry level, or maybe even have it as a configuration thing.